ddtrace/internal/processor/endpoint_call_counter.py

```diff
@@ -0,0 +1,44 @@
+import typing
+
+import attr
+
+from ddtrace.ext import SpanTypes
+from ddtrace.internal import nogevent
+from ddtrace.internal.compat import ensure_str
+from ddtrace.internal.processor import SpanProcessor
+from ddtrace.span import Span
+
+
+EndpointCountsType = typing.Dict[str, int]
+
+
+@attr.s(eq=False)
+class EndpointCallCounterProcessor(SpanProcessor):
+
+    endpoint_counts = attr.ib(init=False, repr=False, type=EndpointCountsType, factory=lambda: {}, eq=False)
+    _endpoint_counts_lock = attr.ib(init=False, repr=False, factory=nogevent.DoubleLock, eq=False)
+    _enabled = attr.ib(default=False, repr=False, eq=False)
+
+    def enable(self):
+        # type: () -> None
+        self._enabled = True
+
+    def on_span_start(self, span):
+        # type: (Span) -> None
+        pass
+
+    def on_span_finish(self, span):
+        # type: (Span) -> None
+        if not self._enabled:
+            return
+        if span._local_root == span and span.span_type == SpanTypes.WEB and span.resource is not None:
+            resource = ensure_str(span.resource, errors="backslashreplace")
```

Comment:
Can we indicate here why this coercion is necessary?

I see this resource coercion elsewhere in the profiling code. I would think this is something we should handle in an integration.

Reply:
@majorgreys Is it true that, if `span.resource` is not `None`, it is an instance of `str`? If so, I think we can omit the coercion here.

Reply:
My main concern would be if it would be possible for an app to set this to something that is not a string.

Reply:
The concern is that we would get not-a-string object. Because later we try to render this to the json.

Reply:
Also, FWIW we have the same logic here: https://github.com/DataDog/dd-trace-py/blob/747c03653f16445aabde9c53d66159cb630bde60/ddtrace/profiling/exporter/pprof.pyx#L667